src/dimension/mod.rs

```diff
@@ -404,17 +405,18 @@ fn to_abs_slice(axis_len: usize, slice: Slice) -> (usize, usize, isize) {
     (start, end, step)
 }
 
-/// This function computes the offset from the logically first element to the first element in
-/// memory of the array. The result is always <= 0.
-pub fn offset_from_ptr_to_memory<D: Dimension>(dim: &D, strides: &D) -> isize {
+/// This function computes the offset from the lowest address element to the
+/// logically first element. The result is always >= 0.
+pub fn offset_from_low_addr_ptr_to_logical_ptr<D: Dimension>(dim: &D, strides: &D) -> usize {
     let offset = izip!(dim.slice(), strides.slice()).fold(0, |_offset, (d, s)| {
         if (*s as isize) < 0 {
```

Comment:
I am thinking about whether it would be better to change the condition to `if (*s as isize) < 0 && *d != 0`. In this way, we can allow the step size to be negative when the Axis value is 0 (theoretically, the step size can be any value at this time), and it can avoid panic in the following case:

    use ndarray::{Array, ShapeBuilder};
    fn main() {
        let s = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13];
        let a= Array::from_shape_vec((2, 0, 2).strides((1, -4isize as usize, 2)),s[0..12].to_vec()).unwrap();
        println!("{:?}", a);
    }

Output:

      [[[]]], shape=[2, 0, 2], strides=[1, -4, 2], layout=Cc (0x5), const ndim=3

Reply:
Good point, this seems like a bug that should be fixed ASAP

Reply:
Yeah, good catch @SparrowLii. I've created #998 to fix this in 0.15.2.